### PR TITLE
Replace `staticmethod` calls with decorators

### PR DESCRIPTION
**Pull request by ZedThree**
Replaces lines like `match = staticmethod(match)` with `@staticmethod` decorator on the corresponding method

### CODE REVIEW

File: src/fparser/two/Fortran2003.py [hunk 1]
```diff
@@ -1097,14 +1100,15 @@ class Complex_Literal_Constant(Base):  # R421
...
+    @staticmethod
     def match(string):
         if not string or string[0]+string[-1] != '()':
             return
         if not pattern.abs_complex_literal_constant.match(string):
             return
         r, i = string[1:-1].split(',')
         return Real_Part(r.strip()), Imag_Part(i.strip())
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:1111]: Please delete this blank line.

File: src/fparser/two/Fortran2003.py [hunk 2]
```diff
@@ -1622,7 +1628,7 @@ def match(reader):
         if content:
             return (content,)
         return None
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:1631]: Please rm this blank line.

File: src/fparser/two/Fortran2003.py [hunk 3]
```diff
@@ -1781,7 +1788,7 @@ def match(string):
         else:
             assert newline == '', repr(newline)
         return name, array_spec, char_length, init
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:1791]: Please rm this blank line.

File: src/fparser/two/Fortran2003.py [hunk 4]
```diff
@@ -2459,7 +2468,7 @@ def match(string):
         ts = repmap(ts)
         line = repmap(line)
         return Type_Spec(ts), Ac_Value_List(line)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:2471]: Please rm this blank line.

File: src/fparser/two/Fortran2003.py [hunk 5]
```diff
@@ -2511,7 +2521,7 @@ def match(string):
         s1 = repmap(line[:j].rstrip())
         s2 = repmap(line[j+1:].lstrip())
         return Ac_Value_List(s1), Ac_Implied_Do_Control(s2)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:2524]: Please rm this line.

File: src/fparser/two/Fortran2003.py [hunk 6]
```diff
@@ -2912,7 +2923,7 @@ def match(string, target=False):
         else:
             assert newline == '', repr((newline, string))
         return name, array_spec, char_length, init
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:2926]: Please rm this line.

File: src/fparser/two/Fortran2003.py [hunk 7]
```diff
@@ -2947,13 +2958,14 @@ class Initialization(Base):  # R506
...
 
+    @staticmethod
     def match(string):
         if string.startswith('=>'):
             return '=>', Null_Init(string[2:].lstrip())
         if string.startswith('='):
             return '=', Initialization_Expr(string[1:].lstrip())
         return None
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:2968]: Please rm this line.

File: src/fparser/two/Fortran2003.py [hunk 8]
```diff
@@ -3021,7 +3034,7 @@ def match(string):
         if not line.startswith('='):
             return
         return (Scalar_Char_Initialization_Expr(line[1:].lstrip()),)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:3037]: Please rm.

File: src/fparser/two/Fortran2003.py [hunk 9]
```diff
@@ -3061,7 +3075,7 @@ def match(string):
         if not lower:
             return
         return Lower_Bound(repmap(lower)), Upper_Bound(repmap(upper))
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:3078]: Please rm.

File: src/fparser/two/Fortran2003.py [hunk 10]
```diff
@@ -3137,7 +3152,7 @@ def match(string):
             return
         line = line[:-1].rstrip()
         return Explicit_Shape_Spec_List(line), None
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:3155]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 11]
```diff
@@ -3256,7 +3273,7 @@ def match(string):
         if not lhs or not rhs:
             return
         return Language_Binding_Spec(lhs), Bind_Entity_List(rhs)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:3276]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 12]
```diff
@@ -3436,7 +3454,7 @@ def match(string):
         if not lhs or not rhs:
             return
         return Data_Stmt_Repeat(lhs), Data_Stmt_Constant(rhs)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:3457]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 13]
```diff
@@ -3506,7 +3525,7 @@ def match(string):
         if not decls:
             return
         return (decls,)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:3528]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 14]
```diff
@@ -3538,7 +3558,7 @@ def match(string):
         if not line:
             return
         return Intent_Spec(spec), Dummy_Arg_Name_List(line)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:3561]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 15]
```diff
@@ -3938,7 +3963,7 @@ def match(string):
         if not ('A' <= lhs <= rhs <= 'Z'):
             return
         return lhs, rhs
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:3966]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 16]
```diff
@@ -4024,7 +4050,7 @@ def match(string):
         if not tmp.items:
             return
         return obj, tmp
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:4053]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 17]
```diff
@@ -4115,7 +4142,7 @@ def match(string):
                 line = line[i:].lstrip()
             items.append((name, lst))
         return (items,)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:4145]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 18]
```diff
@@ -4392,7 +4421,7 @@ def match(string):
         if rhs:
             rhs_obj = Subscript(repmap(rhs))
         return lhs_obj, rhs_obj, stride_obj
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:4424]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 19]
```diff
@@ -4550,7 +4580,7 @@ def match(string):
         if not lower:
             return
         return Lower_Bound_Expr(repmap(lower)), Upper_Bound_Expr(repmap(upper))
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:4583]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 20]
```diff
@@ -4870,10 +4900,11 @@ class Level_2_Unary_Expr(UnaryOpBase):  # R706.c
     subclass_names = ['Add_Operand']
     use_names = []
 
+    @staticmethod
     def match(string):
         return UnaryOpBase.match(
             pattern.add_op.named(), Add_Operand, string)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:4907]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 21]
```diff
@@ -6974,14 +7009,15 @@ class Goto_Stmt(StmtBase):  # R845
...
+    @staticmethod
     def match(string):
         if string[:2].upper() != 'GO':
             return
         line = string[2:].lstrip()
         if line[:2].upper() != 'TO':
             return
         return (Label(line[2:].lstrip()),)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:7020]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 22]
```diff
@@ -7015,7 +7052,7 @@ def match(string):
         if not line:
             return
         return Label_List(lst), Scalar_Int_Expr(line)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:7055]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 23]
```diff
@@ -7043,7 +7081,7 @@ def match(string):
             return
         labels = [Label(l.strip()) for l in labels]
         return (Scalar_Numeric_Expr(line[1:i].strip()),) + tuple(labels)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:7084]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 24]
```diff
@@ -7055,9 +7093,10 @@ class Continue_Stmt(StmtBase, STRINGBase):  # R848
     """
     subclass_names = []
 
+    @staticmethod
     def match(string):
         return STRINGBase.match('CONTINUE', string)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:7099]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 25]
```diff
@@ -7394,7 +7435,7 @@ def match(string):
         if not tmp:
             return
         return Format(repmap(line[:i].rstrip())), Output_Item_List(tmp)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:7438]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 26]
```diff
@@ -9594,7 +9636,7 @@ def match(string):
                 return 'OPERATOR', Local_Defined_Operator(tmp), \
                     Use_Defined_Operator(r)
         return None, Local_Name(lhs), Use_Name(rhs)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:9639]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 27]
```diff
@@ -10520,14 +10565,15 @@ class Alt_Return_Spec(Base):  # R1222
...
+    @staticmethod
     def match(string):
         if not string.startswith('*'):
             return
         line = string[1:].lstrip()
         if not line:
             return
         return (Label(line),)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:10576]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 28]
```diff
@@ -10776,7 +10823,7 @@ def match(string):
         if not line:
             return
         return Result_Name(name), Proc_Language_Binding_Spec(line)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:10826]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 29]
```diff
@@ -10892,7 +10940,7 @@ def match(string):
         if not c1242_valid(prefix, binding_spec):
             return None
         return prefix, name, dummy_args, binding_spec
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:10943]: Please rm

File: src/fparser/two/Fortran2003.py [hunk 30]
```diff
@@ -10981,14 +11029,15 @@ class Return_Stmt(StmtBase):  # R1236
...
+    @staticmethod
     def match(string):
         start = string[:6].upper()
         if start != 'RETURN':
             return
         if len(string) == 6:
             return (None,)
         return (Scalar_Int_Expr(string[6:].lstrip()),)
-    match = staticmethod(match)
+
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:11040]: Please rm

**Comment by arporter** [comment]
Now that we're using Black most of my quibbles will go away once this is brought up to date with master.

**Comment by ZedThree** [comment]
I just force-pushed to rewrite the last merge commit to remove `[skip ci]` from one of the merged commit messages, so that the CI does actually run

**Review by arporter — APPROVED** [review]
All requested changes have been made (thanks).
Will proceed to merge.

**Pull request merged**